BackEndApp/authv1/auth.py

```diff
@@ -53,3 +57,50 @@ def verify(self):
             return True
 
         return False
+
+
+class OTP:
+    def __init__(self, user, otp=None, expire=None, **kwargs):
+        self.user = user
+        self.username = user.get("username")
+        self.otp = otp
+        self.expire = expire
+        self.db = connect()
+        self.collection_name = kwargs.get("collection", "otp")
+        self.collection = self.db[self.collection_name]
+        self.attributes = kwargs
+
+    def create(self, time_delta=5):
+        try:
+            """
+            time_delta: (int) In minutes
+            """
+            self.expire = datetime.now() + timedelta(minutes=time_delta)
+            self.otp = "".join(
+                random.SystemRandom().choice(string.ascii_uppercase + string.digits)
+                for _ in range(6)
+            )
+            doc_otp = {
+                "otp": self.otp,
+                "expire": self.expire,
+                "username": self.user["username"],
+            }
+            self.collection.insert_one(doc_otp)
+            return self.otp
+        except Exception as e:
+            return None
+
+    def find(self):
+        return self.collection.find_one({"username": self.username})
+
+    def verify(self, otp_recieved):
+        user_obj = self.find()
+        stored_otp = user_obj.get("otp")
+        otp_expire = user_obj.get("expire")
+        if otp_expire > datetime.now() and stored_otp == otp_recieved:
+            self.delete()
```

Comment:
```suggestion
        self.delete()
        if otp_expire > datetime.now() and stored_otp == otp_recieved:
```

Reply:
I think doing this would be better, thoughts? @TraXIcoN @shintan777

Reply:
What if they enter the wrong otp. Don't we want to give them another chance

Reply:
If we want the user to have only one chance to verify the otp then this works

Reply:
Hmm valid point @shintan777, but if for some reason, the mail is not sent, then the user would have to wait for 5 mins inevitably  :( 
Won't it be better to just re-send a new OTP in that case? 🤔

Reply:
Valid point, what we can do is check for existing OTP in users name while creating a new one and replace it

Reply:
@shintan777 If for some reason the user requires a new OTP then, I suggest we delete the previously created OTP  object as @ADI10HERO was suggesting, and create a new one for that user because the expiration time for OTP will also have to be updated

Reply:
yes, exactly

Reply:
Okay, so I guess I can commit the requested change right?

Reply:
Delete the otp if its expired, keep it if its incorrect

Reply:
Also if a user requests otp again, it should replace the previous one. Makes sense ? @ADI10HERO

Reply:
So after a lot of discussion, we decided that @TraXIcoN, can commit the suggestion by me, (The user has to re-start the process if something goes wrong in between)